Modify install instructions on how to create a postgres database.

Tomasz-Kluczkowski opened the pull request:
Added necessary steps required to create a postgres database using the standard postgres user account.
Should be useful for users less familiar with postgres database. 
I think we could also add links on installing postgres on main platforms and performing database creation on them as the instructions in my update are for linux.

Tomasz-Kluczkowski committed:
Modify install instructions to show better how to create a postgres database for the project development. Added necessary steps required to create a postgres database using the standard postgres user account.

Tomasz-Kluczkowski committed:
Add project config file to .gitignore.

Added .idea/ folder to .gitignore.

berkerpeksag commented on docs/source/install.rst:
```diff
@@ -56,9 +56,27 @@ And then you'll need to install dependencies::
-    $ createdb pythondotorg -E utf-8 -l en_US.UTF-8
+Change to postgres user to perform postgres database administrative tasks::
+
+    $ sudo su - postgres
```
Out of curiosity, did you get an error when you ran `createdb pythondotorg -E utf-8 -l en_US.UTF-8`? If you need to be "postgres" user, perhaps passing `-O postgres` might work. Could you please try the following command?

```sh
$ createdb pythondotorg -O postgres -E utf-8 -l en_US.UTF-8
```

Tomasz-Kluczkowski replied:
Indeed i get an error:
createdb: database creation failed: ERROR:  permission denied to create database
Even using createdb pythondotorg -O postgres -E utf-8 -l en_US.UTF-8
I have to log into posgres user to be able to create a database.
This is on Ubuntu 18.04 with the postgres 10.3 installation.

berkerpeksag replied:
Thanks for the info. I will make another review tonight, sorry for the delay!

berkerpeksag replied:
I think this can be done a little shorter with the following command:

```sh
$ sudo -u postgres createdb pythondotorg -E utf-8 -l en_US.UTF-8
```

Can you try this command?

Tomasz-Kluczkowski committed:
Add instruction on how to setup sqlite database instead of postgres locally.

Added descriptions on how to replace default postgres database with sqlite.

berkerpeksag commented on docs/source/install.rst:
```diff
@@ -56,9 +56,27 @@ And then you'll need to install dependencies::
-default. Run the following command to create a new database::
+default. Run the following commands to create a new database.
 
-    $ createdb pythondotorg -E utf-8 -l en_US.UTF-8
```
Can we keep this command as default and convert your additional commands to a note?

For example:

```rst
default. Run the following command to create a new database::

    $ createdb pythondotorg -E utf-8 -l en_US.UTF-8

.. note::

   .. You can change the wording, I just use it as an example.

   If you get an error message by running the command above::

       createdb: could not connect to database template1: FATAL:  Peer authentication failed for user "postgres"


   ...

.. Rest of the documentation here.
```

berkerpeksag commented on docs/source/install.rst:
```diff
@@ -68,7 +86,18 @@ variable)::
 
-Now it's time to run migrations::
+If you prefer to use a simpler setup for your database you can use sqlite engine.
+To change database configuration to sqlite go to ``pydotorg/settings/local.py`` and:
```
What do you think about converting this to mention the `DATABASE_URL` environment variable? Contributors can use SQLite without editing `pydotorg/settings/local.py`:

```sh
$ DATABASE_URL="sqlite:///pythondotorg.db" ./manage.py migrate
```

or

```sh
$ export DATABASE_URL="sqlite:///pythondotorg.db"
$ ./manage.py migrate
```

berkerpeksag commented on .gitignore:
```diff
@@ -15,4 +15,5 @@ static-root
-pydotorg/settings/mylocal.py
\ No newline at end of file
+pydotorg/settings/mylocal.py
+.idea/
```
We usually prefer adding things like this to a global .gitignore. See https://help.github.com/articles/ignoring-files/#create-a-global-gitignore for details.

Tomasz-Kluczkowski replied:
Thanks. You never stop learning.

Tomasz-Kluczkowski committed:
Modify install.rst further.

Applied suggestions of @berkerpeksag to the install.rst file.
Simplified sqlite use and added notes suggesting necessary steps to the
users.
Fixed reported no blank line error at the bottom of the file.

berkerpeksag committed:
fix indentation and reflow paragraphs

berkerpeksag committed:
revert .idea addition

berkerpeksag committed:
Update .gitignore
